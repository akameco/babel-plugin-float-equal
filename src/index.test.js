// @flow
import babelTester from 'babel-plugin-tester'
import plugin from '.'

babelTester({
  plugin,
  snapshot: true,
  tests: [
    { title: 'expression', code: `0.1 + 0.2 === 0.3` },
    { title: 'oprator is not ===', code: `0.1 + 0.2 == 0.3` },
    { title: 'expression 2', code: `0.3 === 0.1 + 0.2` },
    {
      title: 'var',
      code: `
        var a = 0.1
        var b = 0.2
        a + b === 0.3
    `
    },
    {
      title: 'valiable 2',
      code: `
        var a = 0.1
        var b = 0.2
        var c = 0.3
        var x = 'hoge'
        a + b === c
        c === a + b
    `
    },
    {
      title: 'not number',
      code: `
        'hello' === 'world'
    `
    },
    {
      title: 'if statement',
      code: `
      if (0.1 + 0.2 === 0.3) {
      }
    `
    }
  ]
})
