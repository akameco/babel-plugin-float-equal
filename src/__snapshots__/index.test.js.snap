// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expression 1`] = `
"
0.1 + 0.2 === 0.3

      ↓ ↓ ↓ ↓ ↓ ↓

typeof (0.1 + 0.2) === 'number' && typeof 0.3 === 'number' ? Math.abs(0.1 + 0.2 - 0.3) < Number.EPSILON : 0.1 + 0.2 === 0.3;
"
`;

exports[`expression 2 1`] = `
"
0.3 === 0.1 + 0.2

      ↓ ↓ ↓ ↓ ↓ ↓

typeof 0.3 === 'number' && typeof (0.1 + 0.2) === 'number' ? Math.abs(0.3 - (0.1 + 0.2)) < Number.EPSILON : 0.3 === 0.1 + 0.2;
"
`;

exports[`if statement 1`] = `
"
if (0.1 + 0.2 === 0.3) {
}

      ↓ ↓ ↓ ↓ ↓ ↓

if (typeof (0.1 + 0.2) === 'number' && typeof 0.3 === 'number' ? Math.abs(0.1 + 0.2 - 0.3) < Number.EPSILON : 0.1 + 0.2 === 0.3) {}
"
`;

exports[`not number 1`] = `
"
'hello' === 'world'

      ↓ ↓ ↓ ↓ ↓ ↓

typeof 'hello' === 'number' && typeof 'world' === 'number' ? Math.abs('hello' - 'world') < Number.EPSILON : 'hello' === 'world';
"
`;

exports[`oprator is not === 1`] = `
"
0.1 + 0.2 == 0.3

      ↓ ↓ ↓ ↓ ↓ ↓

0.1 + 0.2 == 0.3;
"
`;

exports[`valiable 2 1`] = `
"
var a = 0.1
var b = 0.2
var c = 0.3
var x = 'hoge'
a + b === c
c === a + b

      ↓ ↓ ↓ ↓ ↓ ↓

var a = 0.1;
var b = 0.2;
var c = 0.3;
var x = 'hoge';
typeof (a + b) === 'number' && typeof c === 'number' ? Math.abs(a + b - c) < Number.EPSILON : a + b === c;
typeof c === 'number' && typeof (a + b) === 'number' ? Math.abs(c - (a + b)) < Number.EPSILON : c === a + b;
"
`;

exports[`var 1`] = `
"
var a = 0.1
var b = 0.2
a + b === 0.3

      ↓ ↓ ↓ ↓ ↓ ↓

var a = 0.1;
var b = 0.2;
typeof (a + b) === 'number' && typeof 0.3 === 'number' ? Math.abs(a + b - 0.3) < Number.EPSILON : a + b === 0.3;
"
`;
